source/generators/LambertianGenerator.cc

```diff
@@ -0,0 +1,150 @@
+// ----------------------------------------------------------------------------
+// nexus | LED450LWcone.cc
+//
+// This class is a generator for particle events generated in a given cone
+// following a Lambertian distribution. The user must specify via configuration
+// parameters the particle type, a kinetic energy interval, momentum direction
+// and phi and theta limits.
+//
+// The NEXT Collaboration
+// ----------------------------------------------------------------------------
+
+#include "LED450LWcone.h"
+
+#include "DetectorConstruction.h"
+#include "GeometryBase.h"
+#include "RandomUtils.h"
+#include "FactoryBase.h"
+
+#include <G4GenericMessenger.hh>
+#include <G4ParticleDefinition.hh>
+#include <G4RunManager.hh>
+#include <G4ParticleTable.hh>
+#include <G4PrimaryVertex.hh>
+#include <G4Event.hh>
+#include <G4RandomDirection.hh>
+#include <Randomize.hh>
+#include <G4RandomTools.hh>
+#include <G4OpticalPhoton.hh>
+
+#include "CLHEP/Units/SystemOfUnits.h"
+
+using namespace nexus;
+using namespace CLHEP;
+
+REGISTER_CLASS(LED450LWcone, G4VPrimaryGenerator)
+
+
+LED450LWcone::LED450LWcone():
```

Comment:
You should replace `LED450LWcone` with `LambertianGenerator` everywhere in the class.